### PR TITLE
applications: nrf_desktop: Keep HF clock on

### PR DESCRIPTION
**Pull request by pdunaj**

### CODE REVIEW

File: applications/nrf_desktop/src/modules/hfclk_lock.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	}
+}
+
+static void hfclk_unlock(void)
+{
+	if (!hfclk_dev) {
+		return;
+	}
+
+	clock_control_on(hfclk_dev, NULL);
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/hfclk_lock.c]: clock_control_off ?

File: applications/nrf_desktop/src/modules/Kconfig [hunk 2]
```diff
@@ -17,3 +17,4 @@ rsource "Kconfig.config_channel"
 rsource "Kconfig.fn_keys"
 rsource "Kconfig.watchdog"
 rsource "Kconfig.config"
+rsource "Kconfig.hotfix"
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/Kconfig]: Hotfixes - this is suppose to keep multiple fixes needed

**Comment by SebastianBoe** [comment]
Should we send this upstream so more users can use it?

File: applications/nrf_desktop/src/modules/Kconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/Kconfig]: done

File: applications/nrf_desktop/src/modules/hfclk_lock.c (hunk 1 above)
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/hfclk_lock.c]: fixed

**Comment by pdunaj** [comment]
> Should we send this upstream so more users can use it?

I think not. This depends much on what the application does. We don't care so much about slightly higher current consumption as long as user actively works with mouse.
The situation is different when you use the same controller on a device that exchanges data rarely and would like to keep the current consumption on. I separated it as a hotfix as I don't think we will enable it for all our devices.

**Comment by SebastianBoe** [comment]
Ah, I thought this would just keep HF clock on, but I see it keeps it on only when the SW is "powered on", then as you say, this is app-specific.

**Pull request merged**